crates/router/src/utils/verification.rs

```
@@ -0,0 +1,99 @@
+        .change_context(api_error_response::ApiErrorResponse::InternalServerError)?;
+
+    let request_body = verifications::ApplepayMerchantVerificationConfigs {
+        domain_names: body.domain_names.clone(),
+        encrypt_to: APPLEPAY_INTERNAL_MERCHANT_IDENTIFIER.to_string(),
+        partner_internal_merchant_identifier: APPLEPAY_INTERNAL_MERCHANT_IDENTIFIER.to_string(),
+        partner_merchant_name: APPLEPAY_INTERNAL_MERCHANT_NAME.to_string(),
+    };
+
+    let applepay_req = types::RequestBody::log_and_get_request_body(
+        &request_body,
+        utils::Encode::<verifications::ApplepayMerchantVerificationRequest>::encode_to_string_of_json,
+        )
+        .change_context(errors::ApiErrorResponse::InternalServerError)
+        .attach_printable("Failed to encode ApplePay session request to a string of json")?;
+
+    let apple_pay_merch_verification_req = services::RequestBuilder::new()
+        .method(services::Method::Post)
+        .url(APPLEPAY_MERCHANT_VERIFICATION_URL)
+        .attach_default_headers()
+        .headers(vec![(
+            headers::CONTENT_TYPE.to_string(),
+            "application/json".to_string().into(),
+        )])
+        .body(Some(applepay_req))
+        .add_certificate(Some(cert_data))
+        .add_certificate_key(Some(key_data))
+        .build();
+
+    let response = services::call_connector_api(state, apple_pay_merch_verification_req).await;
+    log_applepay_verification_response_if_error(&response);
+
+    let applepay_response =
+        response.change_context(api_error_response::ApiErrorResponse::InternalServerError)?;
+    // Error is already logged
+    Ok(match applepay_response {
+        Ok(_) => services::api::ApplicationResponse::Json(ApplepayMerchantResponse {
+            status_code: "200".to_string(),
+            status_message: "Applepay verification Completed".to_string(),
+        }),
+        Err(_) => services::api::ApplicationResponse::Json(ApplepayMerchantResponse {
```

log the error, don't discard the errors